### DISCUSSION TITLE
Clustering?

### DISCUSSION THREAD

**Discussion by randallc79** [BeryJu/gravity#954] [General]
Kinda new to 'gravity', but trying to figure out how this clustering works?  I have a proxmox ve, with 3 Debian 11 vm's running docker.  Each has a gravity ran from docker-compose file.  In the first server, I've set the "ADMIN_TOKEN", then in the other two, I've set the "ETCD_JOIN_CLUSTER" with IP at the end for the server with the admin_token.  So far I've yet to be able to get any of this clustering to work?

**Comment by daudo** [comment]
I'm currently evaluating gravity, too, so I may not be entirely right about this. But as far as understood it, clustering mostly boils down to using an external etcd. Everything regarding gravity is then stored there and you can easily spawn up any number of nodes that will talk to this single source of truth etcd. The way you did it makes the first gravity node with the ADMIN_TOKEN the single point of failure.

Apart from that, there's not much you need to do. DHCP and DNS requests will be served by any of the running gravity nodes.

And the entire process of actually joining a cluster is described here:

https://gravity.beryju.io/docs/install/cluster/

**Comment by randallc79** [comment]
I'll have to check that out, right now, I'm having to do a complete re-install because outta the blue last night the webui just started refusing connection and the dns / dhcp went down unexpectedly.

**Comment by randallc79** [comment]
well after fighting with it all weekend, I have to go back to my win16 servers for dns / dhcp.  idk, why but even with using etcd, never could get the clustering to work.  not to mention, after setting up the single server and configuring for over 50+ static dhcp entries and dns entries, it would run for about 4 hours and then crash, refusing to hand out dhcp as well as being laggy.  I even tried running it on a single physical box with dual quad procs with 32gb ram each proc.  Still same issue, soon as I entered more that 40 or 50 entries, it'd crash about 4ish hours later.
